tools/Makefile.host

```diff
@@ -131,7 +131,8 @@ endif
 # nxstyle - Check a file for compliance to NuttX coding style
 
 nxstyle$(HOSTEXEEXT): nxstyle.c
-	$(Q) $(HOSTCC) $(HOSTCFLAGS) -o nxstyle$(HOSTEXEEXT) nxstyle.c
+	$(Q) $(HOSTCC) $(HOSTCFLAGS) -DTOPDIR=\"$(realpath $(TOPDIR))\" \
```

Comment:
can we handle the file under apps folder?

Reply:
Oops, I forgot about the apps folder. The implementation is trivial if we could assume the project structure will always be:
- NuttX
  - nuttx
  - apps

Reply:
At least in the current state it won't report false positives.

Reply:
But, can't catch error in apps folder too?:(

Reply:
Right now, to the best of my knowledge, `nxstyle` is only able to reliably report this kind of issue if it knows what is the project root directory.
All of the other algorithms we've discussed so far either have blind spots (**false negatives**) or report false alarms (**false positives**).
I understand that we should avoid accepting a solution that reports **false positives**, because it would mess with the CI reports and create some trouble to the developers.
On the other hand, we could accept **false negatives** to a certain degree, because anything we end up accepting will be better than what we have today (no verification at all).
We just need to agree on the best solution, or maybe someone prove that I am wrong and there is a better solution that is able to address every scenario with no false reports.

Reply:
So, as stupid as it may sound, but regarding files from the `apps` repository, I've just realized that the relative path in their headers all begin with `apps/`. This makes it easier for nxstyle to identify files from `apps` without knowing the project root folder.

I've just updated the PR basically mixing both approaches:
1) If TOPDIR had been defined via Makefile, it will use the most accurate algorithm for files from `nuttx` repository.
2) Otherwise, it will use the algorithm suggested by @xiaoxiang781216 to identify issues with the relative path of files from `apps` repository (and also `nuttx` in case TOPDIR hadn't been defined).

The only currently known limitation is that the approach **2** results in **false negatives** when one or more directories are omitted from the beginning of the relative path (.e.g.: "nxstyle.c" instead of "tools/nxstyle.c").